packages/flutter_tools/lib/src/base/file_system.dart

```diff
@@ -239,7 +239,7 @@ class LocalFileSystem extends local_fs.LocalFileSystem {
       }
       // Make sure that the temporary directory is cleaned up when the tool
       // exits normally.
-      _shutdownHooks?.addShutdownHook(
+      _shutdownHooks.addShutdownHook(
```

Comment:
A test that `_tryToDeleteTemp` runs at some point would be good. I wouldn't be surprised if there's already a test that's trying to do that, and it just wasn't working for some reason.

Reply:
You are right, there were tests for this that didn't verify it, however I think I can think of one now...